src/ModuleManager.cpp

```diff
@@ -484,15 +484,15 @@ void ModuleManager::UnloadModule(ModuleInterface *module)
 {
    if (module)
    {
-      module->Terminate();
-
-      delete module;
-
       if (mLibs.find(module) != mLibs.end())
       {
          mLibs[module]->Unload();
          mLibs.erase(module);
       }
+
+      module->Terminate();
+
+      delete module;
```

Comment:
If you unload the library before you terminate the module, then what will get executed?  The module's code is no longer there...